homeassistant/components/ambient_network/sensor.py

```diff
@@ -266,6 +267,13 @@
         suggested_display_precision=2,
         entity_registry_enabled_default=False,
     ),
+    SensorEntityDescription(
```

Comment:
How is this supposed to work? Can integrations set `last_reported`?

Reply:
The other thing to note is that the time when the Ambient Network reports data is not the same as the time the weather data was collected. See the following screenshot. The weather data was collected on each hour (state) but only reported about 20 minutes later.

![Screenshot 2024-05-28 at 6 34 02 PM](https://github.com/home-assistant/core/assets/542423/39e503b8-0ad7-4aba-80dd-b3193c8390bd)

Reply:
Why is there a 20 min gap?

Reply:
> How is this supposed to work? Can integrations set `last_reported`?

last_reported is set whenever the integration writes the state of an entity

Reply:
> Why is there a 20 min gap?
I do not know why there is a 20 min gap. My guess is that the Ambient Networks polls them once an hour since the data is not provided by native Ambient weather stations.

> `last_reported` is set whenever the integration writes the state of an entity
Then that's not what we need in this case. We want the timestamp of when the measurement was taken, not when the measurement was reported. That's why I introduced a separate entity.

Reply:
I see, but shouldn't that be a state attribute of the sensors and/or weather attributes instead of separate entities?

Reply:
Yes, you are right. That's cleaner. I made it a state attribute instead.